nemo/collections/tts/data/text_to_speech_dataset_lhotse.py

```diff
@@ -0,0 +1,317 @@
+        self.codec_model_downsample_factor = codec_model_downsample_factor
+        self.include_align_prior = prior_scaling_factor is not None
+        self.prior_scaling_factor = prior_scaling_factor
+        self.load_cached_codes_if_available = load_cached_codes_if_available
+        self.dataset_type = dataset_type
+        self.tokenizer_config = tokenizer_config
+        self.load_16khz_audio = load_16khz_audio
+        self.use_text_conditioning_tokenizer = use_text_conditioning_tokenizer
+        self.text_conditioning_tokenizer = None
+        self.pad_context_text_to_max_duration = pad_context_text_to_max_duration
+        self.context_duration_min = context_duration_min
+        self.context_duration_max = context_duration_max
+
+    def __getitem__(self, cuts):
+        cuts = cuts.sort_by_duration()
+
+        logging.debug(f"Len: {len(cuts)}")
+
+        # load audios and text
+        num_codec_frames = []
+        align_priors = []
+        context_audios = []
+        context_audios_lens = []
+        target_audios = []
+        target_audios_lens = []
+        target_audios_16khz = []
+        target_audios_16khz_lens = []
+        context_text_tokens = []
+        context_text_tokens_lens = []
+        has_text_context_list = []
+        target_text_tokens = []
+        target_text_tokens_lens = []
+
+        for i, cut in enumerate(cuts):
+            # load target/answer audio
+            answer_audio = torch.FloatTensor(cut.target_audio.resample(self.sample_rate).load_audio()).squeeze(0)
+            if self.volume_norm:
+                answer_audio = normalize_volume_torch(answer_audio)
+
+            answer_audio = torch.nn.functional.pad(
+                answer_audio,
+                (0, self.codec_model_downsample_factor - (answer_audio.shape[0] % self.codec_model_downsample_factor)),
+                value=0
+            ).unsqueeze(0)
+
+            answer_audio_len = answer_audio.shape[1]
+            target_audios.append(answer_audio)
+            target_audios_lens.append(answer_audio_len)
+            num_frames = int(answer_audio_len / self.codec_model_downsample_factor) + 1 # +1 for EOS
+            num_codec_frames.append(num_frames)
+
+            # load context audio
+            context_audio = torch.FloatTensor(cut.resample(self.sample_rate).load_audio()).squeeze(0)
+            if self.volume_norm:
+                context_audio = normalize_volume_torch(context_audio)
+
+            context_audio = torch.nn.functional.pad(
+                context_audio,
+                (0, self.codec_model_downsample_factor - (context_audio.shape[0] % self.codec_model_downsample_factor)),
+                value=0
+            ).unsqueeze(0)
+            context_audios_len = context_audio.shape[1]
+            context_audios.append(context_audio)
+            context_audios_lens.append(context_audios_len)
+
+            # load context text
+            if cut.supervisions[0].speaker == "user":
```

Comment:
What does speaker == "user" mean here?

Reply:
It is related to the current Lhotse format that we are using for speech-to-speech and that we decided to keep for all other datasets. "user" is input from the user (like context audio/text) and "agent" is what the model will output (target audio and target text). Note that for Speech to speech, the user does not provide the text. For TTS would be more suitable to have the the target text on "user" because it is provided by the user, but we decided to keep the same format.